### DISCUSSION TITLE
🚀 Feature Request: Allow large files during local development

### DISCUSSION THREAD

**Discussion by DylanRJohnston** [cloudflare/workers-sdk#13481] [Feature Requests]
### Describe the solution

Hey all, Wrangler has recently started presenting the following error

```
Error: Asset too large

Cloudflare Workers supports assets with sizes of up to 25MiB. We found a file
[REDACTED].wasm with a size of 240MiB.
Ensure all assets in your assets directory [REDACTED] conform with
the Workers maximum size requirement.
```

The issue is caused by debug wasm builds being very very large as you can see. The final production build is only 3MB but I don't want to throw away all the debug information during local development. Wrangler didn't used to produce this error message for local development and I was wondering if you'd consider removing this restriction or turning it into a warning for local development as development assets tend to be far large than their production counterparts.

**Comment by penalosa** [comment]
Thanks for reporting this! Would you be able to provide a reproduction repo that demonstrates the issue?

**Comment by lrapoport-cf** [comment]
@CarmenPopoviciu @emily-shen  @korinne can you please take a look at this?

**Comment by allsey87** [comment]
I just ran into this too. I guess I will try to downgrade until this disappears :/

This is super annoying for development since it is forcing me to do release builds on my assets when I go to test stuff locally.

**Comment by allsey87** [comment]
Seems like this was merged by @emily-shen in 19b1e698ad209272958a76c27e947a82aad91ca5 on Aug 8, 2024. [wrangler@3.69.1](https://github.com/cloudflare/workers-sdk/releases/tag/wrangler%403.69.1) was released on Aug 6, so hopefully I can migrate to that version without too much pain...

**Comment by petebacondarwin** [comment]
@MattieTK - you were looking into this...
